Firebase/Core/FIRComponentContainer.m

```diff
@@ -92,24 +98,39 @@ - (void)populateComponentsFromRegisteredClasses:(NSSet<Class> *)classes forApp:(
       // Store the creation block for later usage.
       self.components[protocolName] = component.creationBlock;
 
-      // Instantiate the instance if it has requested to be instantiated.
+      // Queue any protocols that should be eagerly instantiated. Don't instantiate them yet
+      // because they could depend on other components that haven't been added to the components
+      // array yet.
       BOOL shouldInstantiateEager =
           (component.instantiationTiming == FIRInstantiationTimingAlwaysEager);
       BOOL shouldInstantiateDefaultEager =
           (component.instantiationTiming == FIRInstantiationTimingEagerInDefaultApp &&
            [app isDefaultApp]);
       if (shouldInstantiateEager || shouldInstantiateDefaultEager) {
-        @synchronized(self) {
-          [self instantiateInstanceForProtocol:component.protocol
-                                     withBlock:component.creationBlock];
-        }
+        [self.eagerProtocolsToInstantiate addObject:component.protocol];
       }
     }
   }
 }
 
 #pragma mark - Instance Creation
 
+- (void)instantiateEagerComponents {
+  // After all components are registered, instantiate the ones that are requesting eager
+  // instantiation.
+  @synchronized(self) {
+    for (Protocol *protocol in self.eagerProtocolsToInstantiate) {
+      // Get an instance for the protocol, which will instantiate it since it couldn't have been
+      // cached yet. Ignore the instance coming back since we don't need it.
+      __unused id unusedInstance = [self instanceForProtocol:protocol];
+    }
+
+    // All eager instantiation is complete, empty and clear the stored property now.
+    [self.eagerProtocolsToInstantiate removeAllObjects];
```

Comment:
Assigning `nil` is sufficient to release all the contents of the array.

Reply:
Changed, thanks.